### PR TITLE
lib: multicell_location: Remove device id

### PR DESCRIPTION
**Pull request by trantanen**
Remove device id parameter from the API.
Reflecting change to multicell_location sample too.

### CODE REVIEW

File: lib/multicell_location/services/skyhook_integration.c [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <modem/lte_lc.h>
 #include <modem/modem_info.h>
```
Comment (jtguggedal) [inline lib/multicell_location/services/skyhook_integration.c:10]: Looks like this can also be removed now, along with any Kconfig dependency

**Review by jtguggedal — APPROVED** [review]
LGTM. Some remnants of modem_info could probably be removed now that IMEI is out of the picture

File: samples/nrf9160/multicell_location/prj.conf [hunk 2]
```diff
@@ -12,6 +12,7 @@ CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
 # Logging
 CONFIG_LOG=y
 CONFIG_LOG_MODE_IMMEDIATE=y
+CONFIG_MULTICELL_LOCATION_LOG_LEVEL_DBG=y
```
Comment (tokangas) [inline samples/nrf9160/multicell_location/prj.conf:15]: Are debug traces enabled intentionally or by mistake?

File: lib/multicell_location/Kconfig [hunk 3]
```diff
@@ -45,9 +45,10 @@ config MULTICELL_LOCATION_SERVICE_HERE
 config MULTICELL_LOCATION_SERVICE_SKYHOOK
 	bool "Skyhook location service"
 	select REST_CLIENT
+	depends on MODEM_JWT
 	help
 	  Use Skyhook location service. If a device ID is not provided for
-	  a location request, the device's IMEI will be used.
+	  a location request, the device's modem UUID will be used.
```
Comment (tokangas) [inline lib/multicell_location/Kconfig:51]: It's no longer possible to provide a device ID, so I guess the second sentence should be removed. The same change should be done also for nRF Cloud service configuration.

File: samples/nrf9160/multicell_location/prj.conf (hunk 2 above)
Comment (trantanen) [reply samples/nrf9160/multicell_location/prj.conf]: Mistake. Removed

File: lib/multicell_location/services/skyhook_integration.c (hunk 1 above)
Comment (trantanen) [reply lib/multicell_location/services/skyhook_integration.c]: Removed.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -73,6 +73,7 @@ nRF9160
   * :ref:`multicell_location` sample:
 
     * Updated to only request neighbor cell measurements when connected and to only copy neighbor cell measurements if they exist.
+    * Remove device ID from :c:func:`multicell_location_get` parameter list. nRF Cloud and HERE didn't use it. Skyhook will now set modem UUID as it's device ID.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:76]:
```suggestion
    * Removed device ID from the :c:func:`multicell_location_get` parameter list. nRF Cloud and HERE did not use it. Skyhook will now set modem UUID as its device ID.
```

File: include/net/multicell_location.h [hunk 5]
```diff
@@ -24,7 +24,8 @@ struct multicell_location {
 	float accuracy;
 };
 
-/* @brief Send a request for location based on cell measurements to the
+/**
+ * @brief Send a request for location based on cell measurements to the
  *        selected location service.
  *
  * @note This function will block until a response
```
Comment (peknis) [inline include/net/multicell_location.h:31]:
```suggestion
 * @note This function will block the calling thread until a response
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**